src/python/tests/ops_smooth.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) 2015-2017 by the parties listed in the AUTHORS file.
+# All rights reserved.  Use of this source code is governed by 
+# a BSD-style license that can be found in the LICENSE file.
+
+from ..mpi import MPI
+from .mpi import MPITestCase
+
+import sys
+import os
+
+import numpy as np
+import healpy as hp
+import libsharp
+
+from ..tod.tod import *
+from ..tod.pointing import *
+from ..tod.sim_tod import *
+from ..tod.sim_det_map import *
+from ..map.pixels import *
+
+
+def expand_pix(startpix, ringpix, local_npix):
+    """Turn first pixel index and number of pixel in full array of pixels
+
+    to be optimized with cython or numba
+    """
+    local_pix = np.empty(local_npix, dtype=np.int64)
+    i = 0
+    for start, num in zip(startpix, ringpix):
+        local_pix[i:i+num] = np.arange(start, start+num)
+        i += num
+    return local_pix
+
+def distribute_rings(nside, rank, n_mpi_processes):
```

Comment:
@tskisner I think this function could be useful for the implementation of the ring-distributed map